### DISCUSSION TITLE
MapLibre and QuadKeys

### DISCUSSION THREAD

**Discussion by sunny4111** [maplibre/maplibre-gl-js#2350] [Q&A]
I want to add bing map which uses QuadKeys in my application, whereas I can't find support for Quadkeys in Maplibre.
If it supports, kindly provide me with link of documentation.

**Comment by HarelM** [comment]
Can you elaborate a bit what you mean? I'm not sure I know what you refer to when you write `Quadkeys`.

**Reply by IvanSanchez** [reply]
@HarelM He means "Bing maps tiles", since Bing is the only (well-known) tile provider which uses quadkeys. See https://github.com/digidem/leaflet-bing-layer/blob/gh-pages/leaflet-bing-layer.js#L7-L24

**Reply by HarelM** [reply]
Thank @IvanSanchez, I guess this is possible using the addProtocol method then...

**Comment by jtmiclat** [comment]
Hi I personally havent tried it but maplibre should already support quadkeys instead of x/y/z.

This following line replaces the `{quadkey}` in url based on the equivalent x/y/z 

https://github.com/maplibre/maplibre-gl-js/blob/7bd37421c51c8dd234880817b167c43407630d82/src/source/tile_id.ts#L42

for example the following source `https://example.com/{x}/{y}/{z}/.pbf` would be written as `https://example.com/{quadkey}.pbf`

**Reply by HarelM** [reply]
Ahh, interesting, this should probably be documented somewhere...

**Comment by barbaramartina** [comment]
Do you know if MapLibre Native supports Quadkeys?

**Reply by HarelM** [reply]
cc: @louwers, can you comment on quadkey support in maplibre native?

**Reply by louwers** [reply]
Supported!
